src/stream.c

```diff
@@ -276,31 +273,30 @@ void avifRWStreamFinishWrite(avifRWStream * stream)
     }
 }
 
-void avifRWStreamWrite(avifRWStream * stream, const void * data, size_t size)
+avifResult avifRWStreamWrite(avifRWStream * stream, const void * data, size_t size)
 {
-    if (!size) {
-        return;
+    if (size) {
+        AVIF_CHECKRES(makeRoom(stream, size));
+        memcpy(stream->raw->data + stream->offset, data, size);
+        stream->offset += size;
     }
-
-    makeRoom(stream, size);
-    memcpy(stream->raw->data + stream->offset, data, size);
-    stream->offset += size;
+    return AVIF_RESULT_OK;
 }
 
-void avifRWStreamWriteChars(avifRWStream * stream, const char * chars, size_t size)
+avifResult avifRWStreamWriteChars(avifRWStream * stream, const char * chars, size_t size)
 {
-    avifRWStreamWrite(stream, chars, size);
+    return avifRWStreamWrite(stream, chars, size);
 }
 
-avifBoxMarker avifRWStreamWriteFullBox(avifRWStream * stream, const char * type, size_t contentSize, int version, uint32_t flags)
+avifResult avifRWStreamWriteFullBox(avifRWStream * stream, const char * type, size_t contentSize, int version, uint32_t flags, avifBoxMarker * marker)
```

Comment:
Optional: This function sets `*marker` to `stream->offset`. Since the caller has access to `stream->offset` (either directly or by calling`avifRWStreamOffset(stream)`), this function doesn't need to provide the `marker` output parameter.

Reply:
`avifRWStreamFinishBox()` requires a marker as argument so it is clearer to have `avifRWStreamWriteBox()` output one in my opinion.

Reply:
Yes, on second thought I agree with you. It is better to treat `marker` as an opaque token that is passed from `avifRWStreamWriteBox()` to `avifRWStreamFinishBox()`.